### DISCUSSION TITLE
Middleware

### DISCUSSION THREAD

**Discussion by securit** [spatie/laravel-package-tools#38] [Ideas]
It would be great if the package tools would allow you to add middleware. One issue I can think of is the order of middleware is important.

**Comment by ousid** [comment]
You can register the middleware directly in your routes, without touching the `kernel.php` file, and if that not the case, you can do that by:

```php
    public function bootingPackage()
    {
        $kernel = $this->app->make(Kernel::class);

        $kernel->appendMiddlewareToGroup('web', YourMiddleware::class);
    }
```

See The Kernel API:
https://laravel.com/api/9.x/Illuminate/Foundation/Http/Kernel.html#method_appendMiddlewareToGroup
